### ISSUE TITLE
✨ Add sorting to "Final decision" column on Talent placement tab

### ISSUE THREAD

**Issue by Jerryescandon** [GCTC-NTGC/gc-digital-talent#9792]
## ✨ Feature

Users would like to be able to sort this column to easily group users by their final decision status.

## 🕵️ Details

The order of statuses should be as following:

Qualified
Qualified pending decision
To assess step 1
To assess step 2
....
Disqualified
Disqualified pending decision
Removed

## 📸 Screenshot

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/16977254/afa104cb-aeaf-43dc-bfaa-fefc07a8a00c)

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] Final decision column is sortable on "Talent placement" table.
- [ ] Final decision column sorting follows the logic described above

## 🛑 Blockers
```[tasklist]
### Blocked By
- [ ] #8966 
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10846
```

**Comment by vd1992** [comment]
@tristan-orourke @esizer 

I'm a little unsure. 
Blocking issue https://github.com/GCTC-NTGC/gc-digital-talent/issues/8966 was to setup the backend sort that enables this issue but I don't think I see it in PR https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642 ?
Or am I missing it somewhere?

**Comment by vd1992** [comment]
The content to sort by is front-generated from what I can tell?
Function `computeInAssessmentStatusChip()`

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/1563aa56-35b6-4c64-bbf7-1795d110a088)

**Comment by vd1992** [comment]
> > I'm a little unsure.
> > Blocking issue #8966 was to setup the backend sort that enables this issue but I don't think I see it in PR #10642 ?
> > Or am I missing it somewhere?
> 
> That was just to move the computation of the needed data to the backend. This ticket is where we use that data to then sort it.

Okay, so this isn't supposed to rely on any backend sorting

**Comment by esizer** [comment]
> Okay, so this isn't supposed to rely on any backend sorting? It is a front-end sort?
> In that case it'd be sorting the current page only?

It is, the data was moved to the back-end in a database column which can now be sorted. The highlighted AC was to ensure that it could be sorted, not to actually sort the data. This ticket is meant to complete the work of actually sorting it on the server and using that in the table.

**Comment by tristan-orourke** [comment]
@esizer actually, rereading those Acceptance Criteria, I think a way of actually sorting it on the server was supposed to be done in #8966, and we forgot 😅.

What was actually completed in #8966 was all about computing the Assessment status on the frontend - which is only one component of the overall "Final decision" status. We'll have to either include that in this ticket now - in which case we should bump up the estimate - or make a new ticket that also blocks this one.
